### PR TITLE
Add auth

### PR DESCRIPTION
**Pull request by okdistribute**

### CODE REVIEW

File: lib/commands/auth/whoami.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+module.exports = {
+  name: 'whoami',
+  command: whoami,
+  options: []
+}
+
+function whoami (opts) {
+  var client = TownshipClient({server: opts.server})
+
+  client.getConfig(function (err, config) {
```
Comment (joehand) [inline lib/commands/auth/whoami.js]: `var login = client.getLogin(opts.server)` should work for this ([docs](https://github.com/township/township-client#townshipgetloginserver)).

File: lib/township.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  var client = TownshipClient(options)
+  var configPath = path.join(homedir(), options.config.filename)
+
+  client.getConfig = function (cb) {
+    fs.readFile(configPath, function (err, contents) {
+      cb(err, JSON.parse(contents.toString()))
+    })
+  }
+
+  client.logout = function (cb) {
```
Comment (joehand) [inline lib/township.js]: I'll add logout to township-client right now! Thanks for the reminder.

File: tests/auth.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+var dat = path.resolve(path.join(__dirname, '..', 'bin', 'cli.js'))
+var baseTestDir = help.testFolder()
+
+var port = process.env.PORT || 3000
+var SERVER = 'http://localhost:' + port
+
+authServer(port, function (server) {
+  test('auth - whoami works when not logged in', function (t) {
+    // cmd: dat clone <link> (same thing again as above)
```
Comment (joehand) [inline tests/auth.js]: We should probably just take all these comments out. I kept making the same mistake copy/pasting.

File: tests/auth.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    var cmd = dat + ' logout'
+    var st = spawn(t, cmd, {cwd: baseTestDir})
+    st.stdout.match(function (output) {
+      t.same('Not logged in.', output, 'output correct')
+      return true
+    })
+    st.stderr.empty()
+    st.end()
+  })
+
```
Comment (joehand) [inline tests/auth.js:30]: We should add a test making sure whoami is empty after logout. Once ready, we'll need to have a secure request test after logout so that can be a placeholder.

**Review by joehand — CHANGES_REQUESTED** [review]
Look good overall! A few small changes and I need to add logout to the client. Should we get auth.dat.land up and going before we release this?

**Comment by joehand** [comment]
`Township-client@1.1.0` has `client.logout({server: opts.server})`!

File: tests/auth.js (hunk 3 above)
Comment (okdistribute) [reply tests/auth.js]: k.

File: tests/auth.js (hunk 4 above)
Comment (okdistribute) [reply tests/auth.js:30]: ok adding

File: lib/commands/auth/whoami.js (hunk 1 above)
Comment (okdistribute) [reply lib/commands/auth/whoami.js]: I'm going to need the `currentLogin` but `getLogin` gets just any old login. I can implement this on the `township-client` side

Comment (okdistribute) [reply lib/commands/auth/whoami.js]: ok I was able to do what i needed by doing `client.config.get()`

**Comment by okdistribute** [comment]
okay changes pushed and rebased to master

**Comment by joehand** [comment]
We should set the config file to be something different for all the auth tests.

**Comment by okdistribute** [comment]
ok pushed changes, fixed the config file path for the tests

**Review by joehand — APPROVED** [review]
